pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -976,6 +976,18 @@ func doActivate(ctx *domainContext, config types.DomainConfig,
 		status.IoAdapterList = config.IoAdapterList
 	}
 
+	if status.IsContainer && config.CloudInitUserData != nil {
```

Comment:
You missed the config.IsCipher part of the check. When the userdata is encrypted CloudInitUserData will be nil per my understanding, since you need to check that one as well.

Reply:
Added